admin/installer/cloudformation.py

```diff
@@ -64,6 +69,33 @@ def _sibling_lines(filename):
     with open(path, 'r') as f:
         return f.readlines()
 
+
+def _get_cluster_size():
+    """
+    Gather desired number of nodes in the cluster from input argument.
+
+    :returns: Desired cluster size, with a default of ``MIN_CLUSTER_SIZE``
+    :rtype: int
+    """
+    parser = argparse.ArgumentParser(
+        description='Create CloudFormation template'
+                    ' for cluster of desired size '
+                    '(default: {0} nodes).'.format(MIN_CLUSTER_SIZE))
+    parser.add_argument('-s', '--size',
+                        default=MIN_CLUSTER_SIZE,
+                        type=int,
+                        help='an integer corresponding to desired '
+                             'number of nodes in the cluster. '
+                             'Supported sizes: min={0} max={1}'.format(
+                                 MIN_CLUSTER_SIZE,
+                                 MAX_CLUSTER_SIZE))
+    size = parser.parse_args().size
+    if size < MIN_CLUSTER_SIZE or size > MAX_CLUSTER_SIZE:
+        raise InvalidClusterSizeException(size)
+    return size
+
+num_nodes = _get_cluster_size()
```

Comment:
:eyes: Would be nice to have all this code in a function rather than at
the top level. It avoid leaking globals in to functions and it not being
caught.